packages/mui-material/src/Badge/Badge.js

```diff
@@ -87,110 +87,153 @@ const BadgeBadge = styled('span', {
     easing: theme.transitions.easing.easeInOut,
     duration: theme.transitions.duration.enteringScreen,
   }),
-  ...(ownerState.color !== 'default' && {
-    backgroundColor: (theme.vars || theme).palette[ownerState.color].main,
-    color: (theme.vars || theme).palette[ownerState.color].contrastText,
-  }),
-  ...(ownerState.variant === 'dot' && {
-    borderRadius: RADIUS_DOT,
-    height: RADIUS_DOT * 2,
-    minWidth: RADIUS_DOT * 2,
-    padding: 0,
-  }),
-  ...(ownerState.anchorOrigin.vertical === 'top' &&
-    ownerState.anchorOrigin.horizontal === 'right' &&
-    ownerState.overlap === 'rectangular' && {
-      top: 0,
-      right: 0,
-      transform: 'scale(1) translate(50%, -50%)',
-      transformOrigin: '100% 0%',
-      [`&.${badgeClasses.invisible}`]: {
-        transform: 'scale(0) translate(50%, -50%)',
+  variants: [
+    {
+      props: ({ ownerState }) => ownerState.color !== 'default',
+      style: {
+        backgroundColor: (theme.vars || theme).palette[ownerState.color].main,
```

Comment:
@brijeshb42 I noticed now that in emotion, we could use the theme and ownerState from the above scope, which is not possible with the zero-runtime. Correct me if I am wrong, but in the zero-runtime this should be:
```suggestion
        backgroundColor: ({ ownerState }) => (theme.vars || theme).palette[ownerState.color].main,
```

Reply:
This this is true, it will make the implementation in emotion a bit more complicated as we will no longer be able to just merge the resolved styles.